qcodes v0.40.0 breaks pytest

With qcodes v0.40.0, our CI (pytest) does not work any more.
PR #54 fixes this temporarily by restricting the qcodes version.

GH actions gives this error message:

```
Run coverage run -m pytest --junit-xml=test-results.xml src/tests
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.2, pluggy-1.3.0
PyQt5 5.15.9 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /home/runner/work/QuMADA/QuMADA
configfile: pyproject.toml
plugins: cases-3.6.14, mock-3.11.1, xvfb-3.0.0, qt-4.2.0
collected 6 items / 1 error

==================================== ERRORS ====================================
__________________ ERROR collecting src/tests/mapping_test.py __________________
ImportError while importing test module '/home/runner/work/QuMADA/QuMADA/src/tests/mapping_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
src/tests/mapping_test.py:44: in <module>
    from qumada.measurement.scripts.generic_measurement import Generic_1D_Sweep
src/qumada/measurement/scripts/__init__.py:21: in <module>
    from .generic_measurement import (
src/qumada/measurement/scripts/generic_measurement.py:33: in <module>
    from qumada.measurement.doNd_enhanced.doNd_enhanced import (
src/qumada/measurement/doNd_enhanced/doNd_enhanced.py:44: in <module>
    from qcodes.dataset.dond.do_nd_utils import (
E   ImportError: cannot import name '_catch_interrupts' from 'qcodes.dataset.dond.do_nd_utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/qcodes/dataset/dond/do_nd_utils.py)
----- generated xml file: /home/runner/work/QuMADA/QuMADA/test-results.xml -----
=========================== short test summary info ============================
ERROR src/tests/mapping_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 2.14s ===============================
Error: Process completed with exit code 2.
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ dynamic = ["version"]
 
 requires-python = ">=3.9"
 dependencies = [
-    "qcodes >= 0.37.0, < 0.40.0",
+    "qcodes >= 0.40.0",
     "qcodes_contrib_drivers >= 0.18.0",
     "matplotlib",
     "jsonschema",

--- a/src/qumada/measurement/doNd_enhanced/doNd_enhanced.py
+++ b/src/qumada/measurement/doNd_enhanced/doNd_enhanced.py
@@ -43,11 +43,11 @@ from qcodes.dataset.descriptions.detect_shapes import detect_shape_of_measuremen
 from qcodes.dataset.descriptions.versioning.rundescribertypes import Shapes
 from qcodes.dataset.dond.do_nd_utils import (
     BreakConditionInterrupt,
-    _catch_interrupts,
     _handle_plotting,
     _register_actions,
     _register_parameters,
     _set_write_period,
+    catch_interrupts,
 )
 from qcodes.dataset.experiment_container import Experiment
 from qcodes.dataset.measurements import Measurement
@@ -184,7 +184,7 @@ def do1d_parallel(
     # do1D enforces a simple relationship between measured parameters
     # and set parameters. For anything more complicated this should be
     # reimplemented from scratch
-    with _catch_interrupts() as interrupted, meas.run() as datasaver, param_meas_caller as call_param_meas:
+    with catch_interrupts() as interrupted, meas.run() as datasaver, param_meas_caller as call_param_meas:
         dataset = datasaver.dataset
         additional_setpoints_data = process_params_meas(additional_setpoints)
 
@@ -348,7 +348,7 @@ def do1d_parallel_asym(
     # do1D enforces a simple relationship between measured parameters
     # and set parameters. For anything more complicated this should be
     # reimplemented from scratch
-    with _catch_interrupts() as interrupted, meas.run() as datasaver, param_meas_caller as call_param_meas:
+    with catch_interrupts() as interrupted, meas.run() as datasaver, param_meas_caller as call_param_meas:
         dataset = datasaver.dataset
         additional_setpoints_data = process_params_meas(additional_setpoints)
 

